tests/test_client.py

```diff
@@ -44,3 +44,32 @@ def test_async_uploader(self):
 
         self.assertIsInstance(async_uploader, AsyncUploader)
         self.assertEqual(async_uploader.client, self.client)
+
+class TusClientTestWithClientCertificate(unittest.TestCase):
+    def setUp(self):
+        self.client_cert=("/tmp/client.crt.pem")
```

Comment:
Does this require the file to be present on the machine? Is there some code, which generates this file?

Reply:
The test only checks if the property is passed along to the uploader. The requests "mock" does not care about it.
Not sure if doing further tests for client certificates in unit tests makes a lot of sense.

Reply:
I see now that this just tests the properties on our classes. Would it be possible to add a test ensuring that the client certificate is properly passed to `requests`? We have similar tests for `verify_tls_cert` and maybe that can serve as an inspiration:

https://github.com/tus/tus-py-client/blob/7b54f7a08add5ac196976c446c288385b8d56510/tests/test_request.py#L54
https://github.com/tus/tus-py-client/blob/7b54f7a08add5ac196976c446c288385b8d56510/tests/test_async_uploader.py#L120

Reply:
I've had a look at the test and also noticed that I forgot to add the certificate for the async upload because there the asyncio library is used for the patches.

Here in uploader: 
https://github.com/tus/tus-py-client/blob/b04cf0b9231f2ef75b1233b63446e8e51471239d/tusclient/uploader/uploader.py#L155

And here in request:  https://github.com/tus/tus-py-client/blob/b04cf0b9231f2ef75b1233b63446e8e51471239d/tusclient/request.py#L112

Need to test if that works and think about how that can be tested. It's getting more complicated than anticipated :/

Reply:
I am sorry to hear that is complicated. Where you able to get any progress on the tests? :)